genrl/bandit/trainer.py

```diff
@@ -30,6 +30,106 @@ def __init__(
         self.log_mode = log_mode
         self.logger = Logger(logdir=logdir, formats=[*log_mode])
 
+    def train(self) -> None:
```

Comment:
For this class (`BanditTrainer`), can you inhertit `abc.ABC` to make it an abstract class. This way, you dont need to raise the NotImplementedError